### DISCUSSION TITLE
bug: Freezes when writing 'hashbang' (#!/)

### DISCUSSION THREAD

**Discussion by herder** [LazyVim/LazyVim#1787] [Q&A]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.9.4

### Operating system/version

Pop!_Os 22.04

### Describe the bug

I was trying to write a bash script, but every time I try to write the `#!/usr/bin/env bash` line at the top the whole UI hard-freezes after `#!`. 

The only way to exit is to kill the whole shell.

I have minimal additions to the standard lazyvim setup - this is what I have added in my `plugins` and `config` dirs:

```
nvim/lua/plugins on  lazyvim [✘!?] via 🌙 v5.1.5 on ☁️
❯ cat navigation.lua tweaks.lua
-- stylua: ignore

return {
  -- add tmux navigator
  -- { "christoomey/vim-tmux-navigator" },
  { "nvim-focus/focus.nvim", version = false,

    keys = {
      {"<leader>wn", "<cmd>FocusSplitNicely<cr>", desc = "Split nicely" },
      {"<leader>wh", "<cmd>FocusSplitLeft<cr>", desc = "Split left" },
      {"<leader>wl", "<cmd>FocusSplitRight<cr>", desc = "Split right" },
      {"<leader>wk", "<cmd>FocusSplitUp<cr>", desc = "Split up" },
      {"<leader>wj", "<cmd>FocusSplitDown<cr>", desc = "Split down" },
    }
}
}
-- stylua: ignore
  -- add melange colorscheme
return {
  { "savq/melange-nvim" },

  -- Configure LazyVim to load gruvbox
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "melange",
    },
  },

}

```

### Steps To Reproduce

1. `nvim test.sh`
2. Start writing `#!/usr/bin/env bash`
3. The UI freezes at `#!`

### Expected Behavior

No freeze :)

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/LazyVim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by herder** [comment]
Note: the bare config above doesn't freeze for me, but I haven't been able to isolate which plugin/config causes the freeze. I'll try to investigate where it fails - the bash lsp support could be a culprit.

**Comment by folke** [comment]
Converted to discussion, since LazyVim is not likely to be the issue here, especially since you can't reproduce it.

If it does turn out to be something special in LazyVim (which I highly doubt), let me know

**Reply by herder** [reply]
Thanks!
I'll see if I can reproduce and get back to you if so.
Thanks for all the great work!

**Comment by herder** [comment]
Everything is working now, so some update somewhere has fixed this for me. Thanks again!
